edb/schema/scalars.py

```diff
@@ -570,10 +570,22 @@ def apply(
             schema = self._validate_enum_change(
                 scls, cur_labels, new_labels, schema, context)
 
+            self.validate_scalar_bases(schema, context)
+
         else:
+            if not self.scls.get_abstract(schema):
+                old_concrete = self.scls.get_topmost_concrete_base(schema)
+
             schema = super().apply(schema, context)
 
-        self.validate_scalar_bases(schema, context)
+            self.validate_scalar_bases(schema, context)
+
+            if not self.scls.get_abstract(schema):
```

Comment:
Why are abstract scalars special?  Wouldn't this loophole let me do

```
abstract scalar type Foo extending str;
scalar type Bar extending Foo;
```

and then `alter scalar type Foo ... extending int64` with bad consequences for `Bar`?  I think we at least ought to check if the abstract scalar has no subtypes.

Reply:
Yeah, you're right. The reason was because abstract scalars might not have a concrete base at all, but that can be handled differently